<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="verbose" default="false" />
  <arg name="visualization" default="true" />
  <arg name="type" default="hex" />
  <arg name="joint_states_topic_name" default="/hydrusx/joint_states" />
  <arg name="launch_model" default="false" />

  <include file="$(find hydrus_transform_control)/launch/hydrusx/hydrusx_bringup.launch" if="$(arg launch_model)">
    <arg name="real_machine" value="false" />
    <arg name="simulation" value="false" />
    <arg name="headless" value="false" />
    <arg name="control_mode" value="0" />
    <arg name="type" value="$(arg type)" />
  </include>
  
  <node name="form_optimization" pkg="hydrus_object_transportation" type="form_optimization_node" respawn="false" output="screen">
    <!-- for transform controller -->
    <rosparam file="$(find hydrus_transform_control)/config/hydrusx/$(arg type)/Hydrus.yaml" command="load" />
    <rosparam file="$(find hydrus_object_transportation)/config/form_optimization.yaml" command="load" />
    <param name="verbose" value="$(arg verbose)" />
    <param name="visualization" value="$(arg visualization)" />
    <param name="joint_states_topic_name" value="$(arg joint_states_topic_name)" />
  </node>
</launch>
